lib/measures/tbd/measure.rb

```diff
@@ -1,210 +1,8 @@
 begin
-  # try to load from the gems
-  $STARTING_DIR = Dir.pwd
-  require "topolys"
-  require "psi"
+  # Try to load from the tbd gem.
+  require "tbd"
 rescue LoadError
-  if $STARTING_DIR != Dir.pwd
-    Dir.chdir($STARTING_DIR)
-  end
-  # load from measure resource dir
-  require_relative "resources/psi.rb"
-  require_relative "resources/conditioned.rb"
-  require_relative "resources/framedivider.rb"
-  require_relative "resources/ua.rb"
-  require_relative "resources/geometry.rb"
-  require_relative "resources/model.rb"
-  require_relative "resources/transformation.rb"
-  require_relative "resources/version.rb"
-  require_relative "resources/log.rb"
-end
-
-##
-# TBD exit strategy. Outputs TBD model content/results if out && io are TRUE.
-# Generates log errors and warnings, even if io or out are FALSE.
-#
-# @param [Model] model OpenStudio model
-# @param [Runner] runner OpenStudio Measure runner
-# @param [Bool] gen_ua True if user wishes to generate UA' metric/report
-# @param [String] ref UA' reference
-# @param [Bool] setpoints True if model zones have valid T° setpoints
-# @param [Bool] out True if user wishes to output detailed TBD content/results
-# @param [Hash] io TBD input/output content
-# @param [Hash] surfaces TBD derated surfaces
-# @param [String] seed OSM file name
-#
-# @return [Bool] Returns true if TBD Measure is successful.
-def exitTBD(model, runner, gen_ua = false, ref = "", setpoints = false, out = false, io = nil, surfaces = nil, seed = "")
-  # Generated files target a design context ( >= WARN ) ... change TBD log_level
-  # for debugging purposes. By default, log_status is set below DEBUG while
-  # log_level is set @WARN. Example: "TBD.set_log_level(TBD::DEBUG)".
-
-  status = TBD.msg(TBD.status)
-  status = TBD.msg(TBD::INFO) if TBD.status.zero?
-
-  unless io && surfaces
-    if TBD.fatal?
-      status = "Halting all TBD processes, halting OpenStudio"
-    else
-      status = "Halting all TBD processes, yet running OpenStudio"
-    end
-  end
-
-  io = {} unless io
```

Comment:
This is a lot of removed code, was this all duplicate code?

Reply:
Nope. I simply moved [exitTBD](https://github.com/rd2/tbd/blob/1e15e4444fbb5f6da99a9a018de5160c2ffebee2/lib/tbd/psi.rb#L3087) at the end of _psi.rb_. This way, _measure.rb_ holds only the ```TBDMeasure``` class. With the exception of a number of ```runner``` calls, _exitTBD_ is more akin to code in _psi.rb_.